[Site Logo]: Add option to remove/clear logo from the block

stacimc opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

Closes #34796, tracked in #30406

## Description
<!-- Please describe what you have changed or added -->
This PR adds a way to clear/remove the media from the Site Logo from within the block. I've added this as ~~a `Remove` button in the block toolbar, similar to `Replace`, an icon in the block toolbar, but would love design feedback.~~ a `Reset` button in the block toolbar, similar to `Replace`, and in keeping with the terminology used by the ToolsPanel.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

I've attempted to test all the edge cases described in #33179 and #32919 for regressions with these new changes.

**IMPORTANT TESTING INSTRUCTIONS: ** Due to an issue raised in #33177, it's tricky to test updates to the actions (basically, the Core php file and the plugin php file for site-logo _both_ get loaded), so you will need to either manually disable the similar code that's running in Core (`wp-includes/blocks/site-logo.php`) or copy over the changes in this PR to that file as well.

Simple flow:
* On a site with no logo, insert a Site Logo block into a post. You should see the placeholder; neither the `Replace` nor `Reset` options are present in the toolbar.
* Upload a logo through the media library. It should replace the placeholder; you should now see the `Replace` and `Reset` options.
* Test that the `Replace` option still works
* Click `Reset`; the placeholder should render again
* Publish the post and refresh. You should still see the placeholder; on the frontend you should see no site logo.

When the logo is persisted:
* Upload a logo through the Site Logo block and save the post; make sure to also save the Site Logo
* Check that you see the Site Logo on the frontend of the post
* In the editor, click `Reset`. You should see the placeholder rendered again.
* Save the post. You should see the option to save the Site Logo as well; make sure this is selected. Refresh the page and the site logo should still display only the placeholder; it should not render on the frontend.

Custom Logo through customizer:
* Set a custom logo in Customizer -> Site Identity and publish. In the block editor, insert a Site Logo block and make sure that it populates with the custom logo you just set.
* Click `Reset` and save the post & Site logo
* Open the Customizer -> Site Identity and verify the logo was removed here as well
* Add a logo through the Site Logo block. Verify it appears in Customizer -> Site Identity.
* Remove the logo from the Customizer, and verify that it is deleted in the block in the block editor as well.

Read-only logos for non-admins:
* Make sure that your test site has a site logo set, either through the block or customizer.
* Create a user with a non-admin role like Author (they should not have the ability to edit the site logo)
* As this user, create a new post and insert a Site Logo block. They should see the site logo, but the `Replace` and `Reset` buttons should not be available. They should still be able to edit block attributes like the rounded style, width, etc.

## Screenshots <!-- if applicable -->
Updated design using 'Reset': 
<img width="370" alt="Screen Shot 2021-09-22 at 11 29 57 AM" src="https://user-images.githubusercontent.com/63313398/134401190-e0847825-70af-4fa5-8067-9a488d7b4f88.png">


Read-only logo with no option to remove/replace, but access to block attrs (design has changed slightly to `Reset`):
<img width="628" alt="Screen Shot 2021-09-14 at 11 12 59 AM" src="https://user-images.githubusercontent.com/63313398/133317492-50c8de61-7fde-45d5-bd86-19c31814732b.png">


## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

stacimc committed:
Add button for removing Site Logo image from block

stacimc commented:
👋 @creativecoder, I've just added you as a reviewer since you've worked on Site Logo recently, but let me know if there's someone else I should ping in addition/instead :)

apeatling commented:
@shaunandrews @jasmussen Should this be an icon for removing an image?

andrewserong commented:
Nice work @stacimc, and thanks again for detailed test instructions, it makes testing so much easier!

The behaviour within the block and full site editing is working well for me, but I ran into a couple of issues with the Customizer vs block settings for the logo. I was testing with TwentyTwenty theme. Here's the steps I used to reproduce:

#### Removing block doesn't appear to override Customizer's setting

1. Activate TwentyTwenty theme
2. From the Customizer, insert a logo
3. Edit a post that contains a Site Logo block
4. Remove the logo and update the post to save the logo
5. Reload the Customizer — the old logo is still there
6. Reload the front end of the site — the old logo is still there

![site-logo-removal-tt-customizer-sml](https://user-images.githubusercontent.com/14988353/133353728-ffab89e7-deac-433e-8f71-fec5bd613d48.gif)

#### Updating the logo from the block, and then removing it reverts the logo back to the one set by the Customizer

1. Activate TwentyTwenty theme
2. From the Customizer, insert a logo
3. Replace the logo with another image and update the post to save the logo
4. Reload the Customizer — the new logo is displayed
5. From the post editor, remove the logo and update the post to save the logo
6. Reload the Customizer — the previous logo set by the Customizer is displayed

![site-logo-removal-tt-customizer-override-sml](https://user-images.githubusercontent.com/14988353/133353768-ec87644d-83d8-46af-9d06-6793bb03df66.gif)

---

> I've added this as a Remove button in the block toolbar, similar to Replace, but would love design feedback.

I'm wondering if having the Remove button in the toolbar could potentially cause confusion for a user wishing to remove the block, but not the site logo altogether?

Would it work to move the Remove button to a sidebar panel where we can have explanatory text that this will remove the logo altogether? I imagine the task of removing the logo image is something that a user would do with less frequency than removing the block itself, so I'd guess having the button be slightly harder to reach for might not be too much of a problem. We could also add an additional Replace button there if it would look strange to have the Remove button on its own.

andrewserong commented:
For the issues I ran into above, I think there's a clue in #33179, which appears to have removed the `_sync_site_logo_to_custom_logo`. To quote @creativecoder:

> Because of this, I propose removing `_sync_site_logo_to_custom_logo`, for now. If we ever have a specific use case (e.g. we need to delete the `custom_logo` setting when the logo is deleted from the block), we could add back exactly what we need, at that time.

If it adds too much complexity to add back in that sync function, I wonder if we can fire an update to the `custom_logo` theme mod separately for this use case?

jasmussen commented:
I've missed this button, thank you for building it. I took the PR for a spin and saw it working as intended with adding and removing the logo from the toolbar:

![site logo](https://user-images.githubusercontent.com/1204802/133391421-8d62876e-56dc-49d0-8608-d5bf74765c94.gif)
_Note, the black underline is an artifact of the theme I'm testing with._

It's mighty convenient and feel like the appropriate place for the button. 

> Should this be an icon for removing an image?

I had the same instinct because we don't yet have a good pattern for two text-only buttons side by side like they are here. I made some quick explorations:

<img width="434" alt="Site logo toolbar" src="https://user-images.githubusercontent.com/1204802/133392272-b174fca5-bb4f-44e2-99b3-1006477bb04b.png">

I don't have a clear favorite of the above, though the simplicity of option 2 is probably a good starting point: it simply increases the spacing between two text-only items. That change might end up being complex, though, due to the gnarly CSS of the block toolbar, and speaking of gap, it suggests we can now refactor the block toolbar to use that for spacing. Worth opening a separate ticket for?

stacimc committed:
Try an icon instead of text button for Remove

stacimc committed:
Delete custom logo when site logo is removed

stacimc commented:
> I ran into a couple of issues with the Customizer vs block settings for the logo.

Thanks for pointing this out @andrewserong, not sure how I missed that in my testing. I've pushed a fix for this, and updated the testing instructions with an important point -- basically you'll need to manually disable the similar site-logo code in Core (`wp-includes/blocks/site-logo.php`) or copy in the new action added to `site-logo/index.php` here.

> I'm wondering if having the Remove button in the toolbar could potentially cause confusion for a user wishing to remove the block, but not the site logo altogether?

I want to say this is probably fine, since the button here doesn't cause the block to be removed at the same time as deleting the media -- so the user gets immediate visual feedback that something else has happened. The change also doesn't persist unless they Publish/Update the post.

> Would it work to move the Remove button to a sidebar panel where we can have explanatory text that this will remove the logo altogether?

Definitely agree that the user should get that explanatory text before potentially making unwanted changes across their site. I think this might be okay though, because when they save a post with modifications to the Site Logo (or likewise in the site editor) they'll get the appropriate warning in the Save message, and be prompted specifically to save the Site Logo separate from the rest of the post:

<img width="273" alt="Screen Shot 2021-09-15 at 4 29 39 PM" src="https://user-images.githubusercontent.com/63313398/133526325-95d44928-54e6-4ffb-b5f8-44e4f7c3977d.png">

I don't have a strong opinion about where it should go -- it seems reasonable to group it with `Replace`, but like you say it's also probably less-commonly-used and could be fine in the Inspector Controls. One benefit there is that it might be easier to nail down the design 😆

For now I've updated the button to an icon as it's the simplest modification, but I can play around with this a bit more to try to wrangle the toolbar styling a bit.

stacimc committed:
Linting

jasmussen commented:
> I don't have a strong opinion about where it should go -- it seems reasonable to group it with Replace, but like you say it's also probably less-commonly-used and could be fine in the Inspector Controls. One benefit there is that it might be easier to nail down the design 😆

It seems reasonable to put it in the inspector as a secondary action. But it would likely need its own share of work in order to ensure the button is supported by enough context to make sense. For example the replace button in the block toolbar works because it's shown in context of the visual itself. So if we were to have a "Remove" button in the inspector, we'd likely want to also show a copy of the image there, complete with a duplicate "replace" button as well — a little like how the Cover block does a focal point picker. 

Happy to create a mockup if need be?

andrewserong commented:
Thanks for the extra context @stacimc and fixing up the issue. I ran out of time to test this again today, but I'll give it another spin tomorrow. I don't feel too strongly about where the Remove button goes, either, and I think your point that the user will still need to confirm when they update the post helps mitigate the concern about the potential danger of it being in the toolbar.

Joen makes a very good point about the additional complexity needed to have the button make sense in the block inspector, so perhaps it's worth keeping it in the toolbar for the time being and we can come back to it as a follow-up if we feel like it could use the extra work?

apeatling commented:
> I don't have a clear favorite of the above, though the simplicity of option 2 is probably a good starting point: it simply increases the spacing between two text-only items

Agree this is a good place to start for this PR.

andrewserong commented on packages/block-library/src/site-logo/edit.js:
```diff
@@ -338,6 +344,7 @@ export default function LogoEdit( {
 				onSelect={ onSelectLogo }
 				onError={ onUploadError }
 			/>
+			<ToolbarButton onClick={ onRemoveLogo } icon={ trash } />
```
I like the use of the trash icon, but should we also add a `label` prop so that we can display a tooltip if the user hovers over the icon?

![image](https://user-images.githubusercontent.com/14988353/133707339-8cb61441-7884-4914-93df-a9d18655f1a1.png)

stacimc replied:
Absolutely -- done!

andrewserong commented on packages/block-library/src/site-logo/index.php:
```diff
@@ -160,3 +160,22 @@ function _delete_site_logo_on_remove_custom_logo_on_setup_theme() {
+function _delete_custom_logo_on_remove_site_logo() {
+	$theme = get_option( 'stylesheet' );
+
+	// Unhook update and delete actions for custom_logo to prevent a loop of hooks.
```
Good idea adding this to avoid accidentally causing a loop of hooks.

✅ Tested that with this code in the Gutenberg plugin, by execution time `has_action` returns `false` so these actions aren't registered, however the code doesn't appear to throw any warnings or cause any issues
✅ Tested that with this code copied to the `site-title.php` file in `public_html/wp-includes/blocks/site-logo.php` the actions _are_ registered, so this code prevents the loop

For testing I added the following and inspected by `debug.log` file:

```
	$has_action_1 = has_action( "update_option_theme_mods_$theme", '_delete_site_logo_on_remove_custom_logo' );
	$has_action_2 = has_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );

	error_log( 'has actions' );
	error_log( var_export( $has_action_1, true ) );
	error_log( var_export( $has_action_2, true ) );
```

The results were:

In Gutenberg plugin:

```
[17-Sep-2021 01:22:46 UTC] has actions
[17-Sep-2021 01:22:46 UTC] false
[17-Sep-2021 01:22:46 UTC] false
```

In Core:

```
[17-Sep-2021 01:36:55 UTC] has actions in core!
[17-Sep-2021 01:36:55 UTC] 10
[17-Sep-2021 01:36:55 UTC] 10
```

creativecoder replied:
Gutenberg's build process is getting in the way here: because `_delete_site_logo_on_remove_custom_logo` and `_delete_site_logo_on_remove_theme_mods` are function names in this file, [they are being automatically prefixed](https://github.com/WordPress/gutenberg/blob/add/remove-site-logo-from-block/tools/webpack/blocks.js#L108) with `gutenberg_`.

This means that the unprefixed Core hooks for the site logo block are not removed, and end up running with `remove_theme_mod( 'custom_logo' )`, the result being that `delete_option( 'site_logo' );` can be called twice. There's [a check](https://github.com/WordPress/wordpress-develop/blob/0f0089391b0f92d4508b8a4df567df2052161340/src/wp-includes/option.php#L738) that prevents this from becoming an infinite loop, but I still think we should prevent this from happening.

You can verify this with the following steps

- Delete any existing site_logo (block) and custom_logo (Customizer) settings
- Add a custom logo by setting the logo in the Customizer
- Insert the Site Logo block in a post/page (you'll see the custom logo you set)
- Reset the logo using the button added in this PR
- If you add a log call or breakpoint to `_delete_site_logo_on_remove_custom_logo` in https://github.com/WordPress/wordpress-develop/blob/0f0089391b0f92d4508b8a4df567df2052161340/src/wp-includes/blocks/site-logo.php#L135, you'll see this function is still called on the `"update_option_theme_mods_$theme"` hook.

Without changing the build process, we could make the relevant function names in this file different from core, and that would work around the issue... something like

```php
function gutenberg_delete_site_logo_on_remove_custom_logo( $old_value, $value ) { ...

function gutenberg_delete_site_logo_on_remove_theme_mods() { ...

function _delete_custom_logo_on_remove_site_logo() {
	$theme = get_option( 'stylesheet' );

	// Unhook update and delete actions for custom_logo to prevent a loop of hooks.
	// Gutenberg hooks.
	remove_action( "update_option_theme_mods_$theme", 'gutenberg_delete_site_logo_on_remove_custom_logo', 10 );
	remove_action( "delete_option_theme_mods_$theme", 'gutenberg_delete_site_logo_on_remove_theme_mods' );

	// Core hooks.
	remove_action( "update_option_theme_mods_$theme", '_delete_site_logo_on_remove_custom_logo', 10 );
	remove_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );

	// Remove the custom logo.
	remove_theme_mod( 'custom_logo' );

	// Restore update and delete actions.
	// Gutenberg hooks.
	add_action( "update_option_theme_mods_$theme", 'gutenberg_delete_site_logo_on_remove_custom_logo', 10, 2 );
	add_action( "delete_option_theme_mods_$theme", 'gutenberg_delete_site_logo_on_remove_theme_mods' );

	// Core hooks.
	add_action( "update_option_theme_mods_$theme", '_delete_site_logo_on_remove_custom_logo', 10, 2 );
	add_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );
}
add_action( 'delete_option_site_logo', '_delete_custom_logo_on_remove_site_logo' );
```

It's a little strange, as those functions will become prefixed with `gutenberg_gutenberg_`, but in practice I don't think that matters.

---

Another option would be to remove `_delete_site_logo_on_remove_custom_logo_on_setup_theme` from this file, and the two functions it hooks in (`_delete_site_logo_on_remove_custom_logo` and `_delete_site_logo_on_remove_theme_mods`).

Because this file is included on the `init` hook, and `_delete_site_logo_on_remove_custom_logo_on_setup_theme` is hooked into `setup_theme`, which runs before `init`, these functions are never run from Gutenberg ([only from Core](https://github.com/WordPress/wordpress-develop/blob/0f0089391b0f92d4508b8a4df567df2052161340/src/wp-includes/blocks/site-logo.php#L162)). They seem to be here only to keep this file consistent with Core. (related issue: https://github.com/WordPress/gutenberg/issues/33177)

stacimc replied:
> There's a check that prevents this from becoming an infinite loop, but I still think we should prevent this from happening.

That explains how I missed it 😱 Thank you for checking this!

I went with the first suggestion to rename the hooks in Gutenberg and unhook/rehook both. I'm a bit torn since I agree it's a little strange, but I think doing so actually makes the hook duplication clearer.

creativecoder replied:
That fixed the issue, thanks!

andrewserong reviewed (commented):
Great work @stacimc! This is testing well for me now:

✅ Simple flow is working correctly 
✅ Logo persistence is testing well in TT1-blocks
✅ Customizer syncing is now working correctly (tested with the change just in Gutenberg, and with the change copied over to core files)
✅ Read-only logos work correctly for non-admins

Personally I quite like the look of adding in the trash icon for the Remove button. I just left a comment on that — if we go with the icon, it could be good to add in a tooltip label.

Once Joen or others are happy with the design and give it the ✅, I think this PR will be good to land.

stacimc committed:
Add label to remove logo button

andrewserong reviewed (approved):
Thanks for adding in the label text @stacimc, this is still testing nicely for me! The code change looks good, so I'm giving it the ✅

I noticed one slight design issue when switching the block editor to use "Display button labels" instead of icons. In that mode, the text size for icon-based buttons is slightly smaller than the normal text size of text-based buttons, so without a separating line the sizes are a tiny bit mismatched:

![image](https://user-images.githubusercontent.com/14988353/133965064-07f723e6-64f9-4658-85fc-a0a09129b5f8.png)

(To set this mode, in the editor go to Options > Preferences > Set Display Button Labels to True)

I imagine that's something at the component level, so we could look into that in a separate PR if it's an issue 🙂

jasmussen commented:
In testing this PR just now, I'm still seeing the trash icon:
<img width="432" alt="Screenshot 2021-09-20 at 10 34 01" src="https://user-images.githubusercontent.com/1204802/133975301-edbd5591-609e-47ee-9f7b-fca2059ba6de.png">

That probably okay if folks feel strongly for that option, but I do have a small preference for starting with option 1 or 2 from [this list of exporations](https://github.com/WordPress/gutenberg/pull/34820#issuecomment-919783072), which just shows "Replace" and "Remove" labels side by side (the full text "Remove logo" feels redundant).

stacimc committed:
Revert to text-based Remove button

stacimc committed:
Slightly increase spacing between Replace and Remove buttons

stacimc commented:
> That probably okay if folks feel strongly for that option, but I do have a small preference for starting with option 1 or 2

Personally I don't have a strong preference, so happy to go back to the text button. I do like the consistency with `Replace`. At the moment I've gone back to the original implementation and slightly increased the space between buttons:
<img width="423" alt="Screen Shot 2021-09-20 at 11 15 06 AM" src="https://user-images.githubusercontent.com/63313398/134053423-7b9d3741-7559-46fb-83eb-4cf894bd395e.png">

Removing the `group` prop from the `BlockControls` gives us something very close to Option 2 with a larger space, but the styling is a teeny bit off -- the `Remove` text is very slightly higher than the `Replace` text:
<img width="415" alt="Screen Shot 2021-09-20 at 9 52 34 AM" src="https://user-images.githubusercontent.com/63313398/134053577-88750b22-d750-495f-ad24-83d6f03be57d.png">

@jasmussen What do you think?

jasmussen commented:
The PR has surfaced a need for some code & design love for multiple text buttons — likely an opportunity to improve how "label only" mode looks as well. I'll make a note to look at it if I get time. But if the 2nd option feels hacky and is visually offset, we should probably go with option 1 for now.

jasmussen commented:
Looking more closely at the screenshot (I'm unable to test in my environment currently) it appears as if the font size is bigger than that of the Image block replace button, so just to be sure, the font size should be 13px same as other buttons. That might also explain the height/offset differential in your second option.

I'm out tomorrow, otherwise I'd be happy to dive in and take a look.

stacimc commented:
> it appears as if the font size is bigger than that of the Image block replace button, so just to be sure, the font size should be 13px same as other buttons.

Font size is 13px for both buttons 🤔  I took another look -- the issue is that there's a `margin-top: -1px` applied to `ToolbarGroup`. So if we wrap just the `Remove` button in `ToolbarGroup` we get this slight offset:
<img width="289" alt="Screen Shot 2021-09-20 at 11 40 30 AM" src="https://user-images.githubusercontent.com/63313398/134056847-909c79aa-5df7-4e42-baeb-d5db925efc3b.png">

If we wrap both buttons in `ToolbarGroup`s, we get the dividing line (Option 3):
<img width="410" alt="Screen Shot 2021-09-20 at 9 52 04 AM" src="https://user-images.githubusercontent.com/63313398/134056978-ce20828e-9757-41b3-9388-0b6300874778.png">

The solution I have right now is to keep the `group` prop on `BlockControls`, so that both buttons are in a `ToolbarGroup`, and then nest the Remove button in an additional `ToolbarGroup` -- that gives us a little bit more spacing between buttons than we'd have otherwise, with equal margins and without the dividing border.

> The PR has surfaced a need for some code & design love for multiple text buttons — likely an opportunity to improve how "label only" mode looks as well.

Agreed! Nesting `ToolbarGroup` in this way to increase spacing feels a bit hacky for now 🤔 I really like your Option 4, for example: 

<img width="476" alt="Screen Shot 2021-09-20 at 11 50 07 AM" src="https://user-images.githubusercontent.com/63313398/134057841-e613aada-a805-42f7-a266-4f518e8f390e.png">

jasmussen commented:
Took the latest branch for a spin, and this is what I see:
<img width="678" alt="Screenshot 2021-09-22 at 08 36 19" src="https://user-images.githubusercontent.com/1204802/134294445-262fd0eb-4353-4296-ae99-c8e6621b0e82.png">

I think it's fine to go with this for now, thanks for the back and forth. It doesn't preclude future iterations, notably the block toolbar and how it's pieced together could use a big refactor to embrace flex and simpler markup/css — but that shouldn't block this nice little improvement from landing.

shaunandrews commented:
I imagine there will be lots of confusion over the word "Remove" and if it relates to the "Remove block" item in the block's ellipsis menu:

<img width="369" alt="image" src="https://user-images.githubusercontent.com/191598/134390656-43778cc8-fe1c-4dfd-8564-5adac98ad755.png">

Perhaps "Reset" would be a better label for this button?

jasmussen commented:
Good point, or "Clear".

shaunandrews commented:
"Reset" matches with the new Tools Panel:

<img width="274" alt="image" src="https://user-images.githubusercontent.com/191598/134396123-59ca5227-cfa8-4ba2-b4e9-9811c76f8cdc.png">

stacimc committed:
Change 'Remove' to 'Reset' to avoid confusion

stacimc commented:
> "Reset" matches with the new Tools Panel

That's a great point -- I'm sold! Changed to `Reset`.

stacimc commented:
@creativecoder I think this is almost good to go; would you mind taking a final look at the syncing of the Site Logo with the custom logo?

simison commented:
One good reason to use "Remove" or "Remove logo" as text label instead of trash-icon is to avoid mixing it with "Remove block":

<img width="237" alt="Screenshot 2021-09-23 at 10 34 33" src="https://user-images.githubusercontent.com/87168/134469995-03b0e7c6-ba9a-4fa9-bb77-4a259ca9cff9.png">

creativecoder reviewed (changes requested):
This is working for me, though has one issue around unregistering Core hooks that I think we should resolve.

stacimc committed:
Unhook both Gutenberg and core hooks for deleting site logo

stacimc committed:
Update comments

creativecoder reviewed (approved):
With the most recent changes, this is working well for me now.

stacimc commented:
@jasmussen Just double checking that this one is ready to merge from a UI perspective with the new `Reset` text?

jasmussen commented:
Yep let's try it!

noisysocks commented on packages/block-library/src/site-logo/index.php:
```diff
@@ -132,7 +132,7 @@ function _sync_custom_logo_to_site_logo( $value ) {
  * @param array $value     Updated theme mod settings.
  */
-function _delete_site_logo_on_remove_custom_logo( $old_value, $value ) {
+function _gutenberg_delete_site_logo_on_remove_custom_logo( $old_value, $value ) {
```
This isn't ideal because it means that, when these changes are merged into Core, that Core will have a function prefixed with `_gutenberg`.

noisysocks commented on packages/block-library/src/site-logo/index.php:
```diff
@@ -160,3 +160,32 @@ function _delete_site_logo_on_remove_custom_logo_on_setup_theme() {
+	add_action( "delete_option_theme_mods_$theme", '_gutenberg_delete_site_logo_on_remove_theme_mods' );
+
+	// Restore Core hooks.
+	add_action( "update_option_theme_mods_$theme", '_delete_site_logo_on_remove_custom_logo', 10, 2 );
+	add_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );
```
`_delete_site_logo_on_remove_custom_logo` and `_delete_site_logo_on_remove_theme_mods` both won't exist when Core's copy of `site-logo.php` [is updated](https://github.com/WordPress/wordpress-develop/pull/1804) to be the `site-logo/index.php` that we see here. This causes a fatal error.

```
Fatal error: Uncaught TypeError: call_user_func_array(): Argument #1 ($callback) must be a valid callback, function "_delete_site_logo_on_remove_custom_logo" not found or invalid function name
```

I think we should revert https://github.com/WordPress/gutenberg/pull/34820/commits/0140ba09a38367bc02215db081b3fb83ff477e25 and work around the issue with Gutenberg's build process in a different way. I'll open a PR to do this.

noisysocks replied:
Fix here: https://github.com/WordPress/gutenberg/pull/36195
